planning/route_handler/src/route_handler.cpp

```diff
@@ -715,12 +715,28 @@ bool RouteHandler::getClosestLaneletWithinRoute(
 bool RouteHandler::getNextLaneletWithinRoute(
   const lanelet::ConstLanelet & lanelet, lanelet::ConstLanelet * next_lanelet) const
 {
+  lanelet::ConstLanelet start_lanelet;
+  lanelet::ConstLanelet goal_lanelet;
+  bool flag_check = true;
```

Comment:
1. It is difficult to understand what does the flag check. Might be better to rename it to a much more expressive name.
2. This can be encapsulate into lambda so that it is more expressive. Example
```
const auto new_name = std::invoke([&](){
if(getGoalLanelet(&goal_lanelet){
return true;
}

if(getClosestLaneletWithinRoute(route_msg_.start_pose, &start_lanelet)){
return true;
}

return goal_lanelet.id() != start_lanelet.id();
}); 
```
Note: my logic might be wrong, so please recheck

Reply:
Yes,I add the "flag_check" to separate the two situations:
(1) It is the problem which the pr shall solve:
![屏幕截图 2023-01-18 132954](https://user-images.githubusercontent.com/102840938/217465466-f33f8247-d576-47e9-b853-2ac7ae61ce95.png)


(2)The second situation:
![image](https://user-images.githubusercontent.com/102840938/217465341-563352e6-bcdf-49be-a9d8-48cffc45f0ee.png)


In the future, the loop planning will be realized as said in this https://github.com/autowarefoundation/autoware.universe/pull/2424

Reply:
thank you for the explanation.

Then i think it is fine if the `flag_check` related conditions is wrapped in a lambda, unless you have additional concerns with it.